Makefile

```diff
@@ -8,25 +8,20 @@ PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 CFLAGS ?= -Wall -Og -g
 
-all:
-	$(MAKE) -C mini
-	$(MAKE) $(PROG)
+all: $(PROG)
 
-mini/ministub.exe:
+mini/ministub.exe: mini/ministub.c
```

Comment:
I think this is changing the recursive
setup to a flat one. In recursive setup,
you are not supposed to pick into subdirs
directly. Instead, you should descend
there with `$(MAKE) -C subdir`, and there
sub-make should check if the update is needed.
Flat setup, OTOH, is when you bubble-up
the deps into top-level. Those techniques
should not be mixed.

Rest looks OK.

Reply:
How about something like this:
```
diff --git a/Makefile b/Makefile
index a4fa286..3f437cc 100644
--- a/Makefile
+++ b/Makefile
@@ -8,11 +8,11 @@ PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 CFLAGS ?= -Wall -Og -g
 
-all:
-       $(MAKE) -C mini
-       $(MAKE) $(PROG)
+all: $(PROG)
+
+.PHONY: force
 
-mini/ministub.exe:
+mini/ministub.exe: force
        $(MAKE) -C mini
 
 stub.exe: mini/ministub.exe
```

Seems to work.

Reply:
I ended up with something similar:

```make
all: $(PROG)

.PHONY: mini
mini:
	$(MAKE) -C mini

mini/ministub.exe: mini

.SECONDEXPANSION:
stub.exe: mini/ministub.exe
	cp $< $@
```

Secondary expansion is necessary for `stub.exe`: `make` needs to re-evaluate the prerequisite, otherwise it might not realise that `ministub.exe` has been updated after running `$(MAKE) -C mini`.

Reply:
I suspect in your case it doesn't realize
this because the recipe for `ministub.exe`
is empty.
Can you check if with my change the
same problem exists?

Reply:
Ah yes, that fixes it.